docs.txt

```diff
@@ -0,0 +1,16 @@
+colorcet
+jinja2
+networkx
+numpy
+packaging
+pandas
+pillow
+pscript
+python-dateutil
+pyyaml
+scipy
+setuptools
+six
+tornado
+
+sphinx>1.8
```

Comment:
Is a text file the only option for specifying all these? Right now we have a single source of truth for all of our build/test/run dependencies in the conda recipe `meta.yaml`. I'd really hate to have to go back to duplicating information in different places it is always a maintenance hassle

Reply:
Using a `requirements.txt` file is the standard way of how Python packages specify their dependencies. Read the Docs follows that pattern and uses `pip` to install them.

Dependencies for the documentation, tests, lint, build, etc are usually different. That's why I created a new file for docs requirements.

I don't have too much knowledge about conda. Is this the file that you mention? https://github.com/bokeh/bokeh/blob/36db0337d0f459bd44c3642e4f06d7f284246b7f/conda.recipe/meta.yaml

Read the Docs supports `conda` on its building process, but we don't usually recommend to use it unless it's strictly required. See https://docs.readthedocs.io/en/stable/guides/conda.html

Reply:
Right, we specify *runtime* dependencies in `setup.py` an then both `meta.yaml` and `requirements.txt` load from that single source of true. 

For test, dev dependencies, we're definitely 100% conda-based for recommended local dev usage. After a long effort we finally have a process that most new users can attempt to do and have a high chance of success. I don't care so much if RTD uses pip or conda but I am concerned about finding ways to avoid duplicating information in multiple places.

Reply:
FYI the way we normally install test/dev deps is:
```
conda install `python scripts/deps.py test`
```
from the top level

Reply:
Bokeh dependency system is currently too custom for any solution to work on Read the Docs. Unfortunately, we didn't find a solution that does not involves a `requirements.txt` file (`docs.txt` in this case).

In the future, Read the Docs will support to install these docs dependencies directly from the `.readthedocs.yml` file (See https://github.com/readthedocs/readthedocs.org/issues/5572), which still won't support how Bokeh manages its dependencies, but at least it will be better for this case.

Let's use a `docs.txt` file for now, and we should revisit this when Read the Docs has support for arbitrary package name installation.